### DISCUSSION TITLE
Recovery operations block all client IOPS, reducing recovery parameter's has no effect.

### DISCUSSION THREAD

**Discussion by Trevor-K-Smith** [rook/rook#12521] [General]
Setting the following appears to have no effect in slowing down recovery operations. I cant't get any Client IO when recovery operation active.

```
ceph config set osd osd_max_backfills 1
ceph config set osd osd_recovery_max_active 1
ceph config set osd osd_recovery_max_active 1
ceph config set osd osd_recovery_max_active_hdd 1
ceph config set osd osd_client_op_priority 63
ceph config set osd osd_recovery_op_priority 0
```

Similar mentions here in rook ceph - https://www.reddit.com/r/ceph/comments/11rby7i/ceph_recovery_misplaced_destroys_my_io_low/

Any guidance be appreciated. Anything i have to restart after setting the above?

**Comment by travisn** [comment]
@rzarzynski Any guidance on what might be missed for adjusting the recovery?

**Reply by rzarzynski** [reply]
@travisn: Hmm, since long time the default for `osd_max_backfills` is `1`. Also `osd_client_op_priority=63` changes nothing (`63` is default in main); you could try po poke with it but it's rather a trial-and-error thing (the WPQ scheduler, sorry).

How about switching to mClock?

You might also want to triangulate with @sseshasa@redhat.com (the `@` mention doesn't work, unfortunately)  who is our QoS Guy.
